sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -620,6 +622,12 @@ class SparkSession private(
    */
   @transient lazy val catalog: Catalog = new CatalogImpl(self)
 
+  @transient private lazy val catalogs = new mutable.HashMap[String, CatalogPlugin]()
+
+  private[sql] def catalog(name: String): CatalogPlugin = synchronized {
+    catalogs.getOrElseUpdate(name, Catalogs.load(name, sessionState.conf))
```

Comment:
Looks like we don't support unloading or reloading a catalog plugin?

Reply:
do you have any use case of unloading/reloading?

Reply:
Possible case like loading a catalog plugin with different config? Currently it has to load it with a different catalog name. For example, like for a REST-based catalog endpoint, if it requires some authentication. We might want to reconnect the catalog with different authentication.

Reply:
I would be fine adding this feature if it is needed, but we don't currently support reconfiguration in SessionCatalog so I think it makes sense to get this in and consider adding it later.